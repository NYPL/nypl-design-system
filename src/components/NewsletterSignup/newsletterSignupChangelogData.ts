/** This data is used to populate the ComponentChangelogTable component.
 *
 * date: string (when adding new entry during development, set value as "Prerelease")
 * version: string (when adding new entry during development, set value as "Prerelease")
 * type: "Bug Fix" | "New Feature" | "Update";
 * affects: array["Accessibility" | "Documentation" | "Functionality" | "Styles"];
 * notes: array (will render as a bulleted list, add one array element for each list element)
 */
import { ChangelogData } from "../../utils/ComponentChangelogTable";

export const changelogData: ChangelogData[] = [
  {
    date: "Prerelease",
    version: "Prerelease",
    type: "Update",
    affects: ["Styles"],
    notes: ["Chakra 2.8 update."],
  },
  {
    date: "2023-11-09",
    version: "2.1.2",
    type: "Update",
    affects: ["Styles"],
    notes: [
      "Updates the dark mode color variants for the `newsletterSignupType` prop.",
    ],
  },
  {
    date: "2023-10-26",
    version: "2.1.1",
    type: "Update",
    affects: ["Accessibility"],
    notes: [
      "Updates `tabindex` value from 0 to -1. See Accessibility section for details.",
    ],
  },
  {
    date: "2023-10-18",
    version: "2.1.0",
    type: "New Feature",
    affects: ["Accessibility", "Documentation", "Functionality", "Styles"],
    notes: ["Adds the NewsletterSignup component to the DS library"],
  },
];
