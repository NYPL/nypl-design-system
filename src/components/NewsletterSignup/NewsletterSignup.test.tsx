import { render, screen } from "@testing-library/react";
//import userEvent from "@testing-library/user-event";
import { axe } from "jest-axe";
import * as React from "react";
import renderer from "react-test-renderer";

import NewsletterSignup from "./NewsletterSignup";
import { sectionDataMap } from "../../helpers/types";

describe("NewsletterSignup Accessibility", () => {
  const onSubmit = jest.fn();
  it("Form state passes accessibility", async () => {
    const { container } = render(
      <NewsletterSignup
        title="Testing"
        onSubmit={onSubmit}
        formHelperText={"Form helper."}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("Error state passes accessibility", async () => {
    const { container } = render(
      <NewsletterSignup title="Testing" onSubmit={onSubmit} view={"error"} />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("Confirmation state passes accessibility", async () => {
    const { container } = render(
      <NewsletterSignup
        title="Testing"
        onSubmit={onSubmit}
        view={"confirmation"}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("Bad email state passes accessibility", async () => {
    const { container } = render(
      <NewsletterSignup
        title="Testing"
        onSubmit={onSubmit}
        view={"form"}
        isInvalidEmail={true}
        formHelperText={"That is not allowed."}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("NewsletterSignup Unit Tests", () => {
  // If you want to see what's happening, insert below render()
  // screen.debug();
  let onSubmit = jest.fn();
  describe("Renders the Minimum Required Elements", () => {
    it("Renders the form", () => {
      render(<NewsletterSignup title="Testing" onSubmit={onSubmit} />);
      expect(screen.getByRole("form")).toBeInTheDocument();
    });
    it("Renders the input textbox", () => {
      render(<NewsletterSignup title="Testing" onSubmit={onSubmit} />);
      expect(screen.getByRole("textbox")).toBeInTheDocument();
    });
    it("Renders the button", () => {
      render(<NewsletterSignup title="Testing" onSubmit={onSubmit} />);
      expect(
        screen.getByRole("button", { name: "Submit" })
      ).toBeInTheDocument();
    });
    it("Renders the title", () => {
      render(<NewsletterSignup title="Testing" onSubmit={onSubmit} />);
      expect(screen.getByText(/Testing/i)).toBeInTheDocument();
    });
  }); // Close minimum elements tests

  describe("Renders the Optional descriptionText and formHelperText Values", () => {
    // Note: newsletterSignupType tests are covered in the snapshot tests below.
    const testNewsletterSignup = (
      <NewsletterSignup
        descriptionText="Do not send cash."
        formHelperText={"Just trying to help"}
        title="Testing"
        onSubmit={onSubmit}
      />
    );
    it("Renders the description", () => {
      render(testNewsletterSignup);
      expect(screen.getByText(/Do not send cash./i)).toBeInTheDocument();
    });

    it("Renders the helper text", () => {
      render(testNewsletterSignup);
      expect(screen.getByText(/Just trying to help/i)).toBeInTheDocument();
    });
  }); // Close optional text elements test

  describe("Renders the Feedback Views", () => {
    it("Renders the error view", () => {
      render(
        <NewsletterSignup view={"error"} title="Testing" onSubmit={onSubmit} />
      );
      expect(
        screen.getByText(/Oops! Something went wrong./i)
      ).toBeInTheDocument();
      expect(screen.getByTitle("errorFilled icon")).toBeInTheDocument();
    });

    it("Renders the confirmation view", () => {
      render(
        <NewsletterSignup
          view={"confirmation"}
          title="Testing"
          onSubmit={onSubmit}
          confirmationText={"Fantastic!"}
        />
      );
      expect(screen.getByText(/Fantastic/i)).toBeInTheDocument();
      expect(
        screen.getByTitle("actionCheckCircleFilled icon")
      ).toBeInTheDocument();
    });

    it("Renders the bad email view", () => {
      render(
        <NewsletterSignup
          view={"form"}
          isInvalidEmail={true}
          title="Testing"
          onSubmit={onSubmit}
        />
      );
      expect(
        screen.getByText(/Please enter a valid email address./i)
      ).toBeInTheDocument();
      expect(screen.getByRole("form")).toBeInTheDocument();
      expect(screen.getByRole("textbox")).toBeInTheDocument();
      expect(screen.getByRole("button")).toBeInTheDocument();
    });
  }); // Close feedback tests

  // @todo These will likely need to be rewritten under the new specs
  // describe("Behaviors on Button Click", () => {
  //   it("Disables the button when clicked", () => {
  //     render(<NewsletterSignup title="Testing" onSubmit={onSubmit} />);
  //     //screen.debug();
  //     const button = screen.getByRole("button", { name: "Submit" });
  //     expect(button).not.toHaveAttribute("disabled");
  //     button.click();
  //     //screen.debug();
  //     expect(button).toHaveAttribute("");
  //   });
  //
  //   it("Adds hidden fields data to the submitted data", () => {
  //     const hiddenFields = {
  //       "hidden-field-1": "hidden-field-value-1",
  //       "hidden-field-2": "hidden-field-value-2",
  //     };
  //     let submittedValues;
  //     let onSubmit = (values) => {
  //       submittedValues = values;
  //     };
  //     render(
  //       <NewsletterSignup
  //         hiddenFields={hiddenFields}
  //         title="Testing"
  //         onSubmit={onSubmit}
  //       />
  //     );
  //
  //     const emailField = screen.getByLabelText(/email/i);
  //     userEvent.type(emailField, "email@email.com");
  //
  //     const button = screen.getByRole("button", { name: "Submit" });
  //
  //     button.click();
  //
  //     expect(submittedValues).toEqual({
  //       email: "email@email.com",
  //       "hidden-field-1": "hidden-field-value-1",
  //       "hidden-field-2": "hidden-field-value-2",
  //     });
  //   });
  // }); // Close behaviors on button click
}); // Close unit tests.

describe("NewsletterSignup Snapshots", () => {
  const onSubmit = jest.fn();
  it("Renders the default form UI snapshot correctly", () => {
    const view = renderer
      .create(<NewsletterSignup title="Testing" onSubmit={onSubmit} />)
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it("Renders the form UI with formHelperText snapshot correctly", () => {
    const view = renderer
      .create(
        <NewsletterSignup
          title="Testing"
          onSubmit={onSubmit}
          formHelperText={"You need help."}
        />
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it("Renders the form UI with description snapshot correctly", () => {
    const view = renderer
      .create(
        <NewsletterSignup
          title="Testing"
          onSubmit={onSubmit}
          descriptionText={"Why not?"}
        />
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it("Renders the bad email UI snapshot correctly", () => {
    const view = renderer
      .create(
        <NewsletterSignup
          title="Testing"
          onSubmit={onSubmit}
          isInvalidEmail={true}
        />
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it("Renders the error UI snapshot correctly", () => {
    const view = renderer
      .create(
        <NewsletterSignup title="Testing" onSubmit={onSubmit} view={"error"} />
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it("Renders the default confirmation UI snapshot correctly", () => {
    const view = renderer
      .create(
        <NewsletterSignup
          title="Testing"
          onSubmit={onSubmit}
          view={"confirmation"}
        />
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it("Renders the confirmation UI with confirmationText snapshot correctly", () => {
    const view = renderer
      .create(
        <NewsletterSignup
          title="Testing"
          onSubmit={onSubmit}
          view={"confirmation"}
          confirmationText={"You did great!"}
        />
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  describe("Renders each color for each newsletterSignupType correctly", () => {
    // newsletterSignupType values are determined by the types contained in the sectionDataMap. So it is safe to use the map directly.
    sectionDataMap.map((section) => {
      it(
        "Renders " +
          section.type +
          " color band with " +
          section.colorVals.primary,
        () => {
          const view = renderer
            .create(
              <NewsletterSignup
                title="Testing"
                onSubmit={onSubmit}
                newsletterSignupType={section.type}
              />
            )
            .toJSON();
          expect(view).toMatchSnapshot();
        }
      );
    });
  }); // Close colors snapshots
}); // Close snapshots
