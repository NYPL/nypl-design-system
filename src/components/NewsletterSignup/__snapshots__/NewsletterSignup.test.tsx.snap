// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsletterSignup Snapshots Renders the bad email UI snapshot correctly 1`] = `
<div
  className="chakra-stack css-1f0wxn3"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <form
      aria-label="form"
      className="css-1xdhyk6"
      id="newsletter-form"
      onSubmit={[MockFunction]}
    >
      <div
        className="css-fadkce"
        id="newsletter-form-parent"
      >
        <div
          className="css-l5w1vc"
          id="newsletter-form-child0"
        >
          <div
            className="css-1u8qly9"
            id="email-input-wrapper"
          >
            <label
              className="css-1xdhyk6"
              htmlFor="email-input"
              id="email-input-label"
            >
              Email Address
              <span>
                 (Required)
              </span>
            </label>
            <div
              className="css-79elbk"
            >
              <input
                aria-describedby="email-input-helperText"
                aria-invalid={true}
                aria-required={true}
                autoComplete="email"
                className="chakra-input css-0"
                disabled={false}
                id="email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your email address"
                required={true}
                step={null}
                type="email"
                value=""
              />
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={true}
              id="email-input-helperText"
            >
              <div
                className="css-0"
                dangerouslySetInnerHTML={
                  {
                    "__html": "There was a problem. Please enter a valid email address.",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="css-l5w1vc"
          id="newsletter-form-child1"
        >
          <button
            className="chakra-button css-1xdhyk6"
            id="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the custom error state snapshot correctly 1`] = `
<div
  className="chakra-stack css-8g8ihq"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <div
      className="feedback-body response css-6cjp67"
      tabIndex={-1}
    >
      <div
        className="css-116fi6s"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-cr4d1u"
          focusable={false}
          role="img"
          title="errorFilled icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <p
          className="chakra-text css-ia9jis"
        >
          Oops! Something went wrong.
        </p>
      </div>
      <div>
        If the error persists,
        <a
          href="mailto:enews@nypl.org?subject=Please re-activate my e-mail address"
        >
          contact our e-mail team
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the default confirmation state snapshot correctly 1`] = `
<div
  className="chakra-stack css-1f0wxn3"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <div
      className="feedback-body response css-6cjp67"
      tabIndex={-1}
    >
      <div
        className="css-116fi6s"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-1grhd2q"
          focusable={false}
          role="img"
          title="actionCheckCircleFilled icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <p
          className="chakra-text css-1e5ycxp"
        >
          Thank you for signing up!
        </p>
      </div>
      <p
        className="chakra-text css-1xdhyk6"
      >
        You can update your email subscription preferences at any time using the links at the bottom of the email.
      </p>
    </div>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the default error state snapshot correctly 1`] = `
<div
  className="chakra-stack css-1f0wxn3"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <div
      className="feedback-body response css-6cjp67"
      tabIndex={-1}
    >
      <div
        className="css-116fi6s"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-cr4d1u"
          focusable={false}
          role="img"
          title="errorFilled icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <p
          className="chakra-text css-ia9jis"
        >
          Oops! Something went wrong.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the default form UI snapshot correctly 1`] = `
<div
  className="chakra-stack css-8g8ihq"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <form
      aria-label="form"
      className="css-1xdhyk6"
      id="newsletter-form"
      onSubmit={[MockFunction]}
    >
      <div
        className="css-fadkce"
        id="newsletter-form-parent"
      >
        <div
          className="css-l5w1vc"
          id="newsletter-form-child0"
        >
          <div
            className="css-1u8qly9"
            id="email-input-wrapper"
          >
            <label
              className="css-1xdhyk6"
              htmlFor="email-input"
              id="email-input-label"
            >
              Email Address
              <span>
                 (Required)
              </span>
            </label>
            <div
              className="css-79elbk"
            >
              <input
                aria-describedby="email-input-helperText"
                aria-required={true}
                autoComplete="email"
                className="chakra-input css-0"
                disabled={false}
                id="email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your email address"
                required={true}
                step={null}
                type="email"
                value=""
              />
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={false}
              id="email-input-helperText"
            >
              <div
                className="css-0"
              >
                Ex: 
                jdoe@domain.com
                <br />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-l5w1vc"
          id="newsletter-form-child1"
        >
          <button
            className="chakra-button css-1xdhyk6"
            id="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the form UI with description snapshot correctly 1`] = `
<div
  className="chakra-stack css-8g8ihq"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <p
      className="chakra-text css-1xdhyk6"
    >
      Why not?
    </p>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <form
      aria-label="form"
      className="css-1xdhyk6"
      id="newsletter-form"
      onSubmit={[MockFunction]}
    >
      <div
        className="css-fadkce"
        id="newsletter-form-parent"
      >
        <div
          className="css-l5w1vc"
          id="newsletter-form-child0"
        >
          <div
            className="css-1u8qly9"
            id="email-input-wrapper"
          >
            <label
              className="css-1xdhyk6"
              htmlFor="email-input"
              id="email-input-label"
            >
              Email Address
              <span>
                 (Required)
              </span>
            </label>
            <div
              className="css-79elbk"
            >
              <input
                aria-describedby="email-input-helperText"
                aria-required={true}
                autoComplete="email"
                className="chakra-input css-0"
                disabled={false}
                id="email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your email address"
                required={true}
                step={null}
                type="email"
                value=""
              />
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={false}
              id="email-input-helperText"
            >
              <div
                className="css-0"
              >
                Ex: 
                jdoe@domain.com
                <br />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-l5w1vc"
          id="newsletter-form-child1"
        >
          <button
            className="chakra-button css-1xdhyk6"
            id="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the form UI with formHelperText snapshot correctly 1`] = `
<div
  className="chakra-stack css-1f0wxn3"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <form
      aria-label="form"
      className="css-1xdhyk6"
      id="newsletter-form"
      onSubmit={[MockFunction]}
    >
      <div
        className="css-fadkce"
        id="newsletter-form-parent"
      >
        <div
          className="css-l5w1vc"
          id="newsletter-form-child0"
        >
          <div
            className="css-1u8qly9"
            id="email-input-wrapper"
          >
            <label
              className="css-1xdhyk6"
              htmlFor="email-input"
              id="email-input-label"
            >
              Email Address
              <span>
                 (Required)
              </span>
            </label>
            <div
              className="css-79elbk"
            >
              <input
                aria-describedby="email-input-helperText"
                aria-required={true}
                autoComplete="email"
                className="chakra-input css-0"
                disabled={false}
                id="email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your email address"
                required={true}
                step={null}
                type="email"
                value=""
              />
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={false}
              id="email-input-helperText"
            >
              <div
                className="css-0"
              >
                Ex: 
                jdoe@domain.com
                <br />
                You need help.
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-l5w1vc"
          id="newsletter-form-child1"
        >
          <button
            className="chakra-button css-1xdhyk6"
            id="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`NewsletterSignup Snapshots Renders the submitting state snapshot correctly 1`] = `
<div
  className="chakra-stack css-8g8ihq"
>
  <div
    className="chakra-stack newsletter-signup-appeal css-1ey9w6j"
  >
    <h2
      className="chakra-heading css-sktjnm"
    >
      Sign Up for Our Newsletter
    </h2>
    <a
      className="chakra-link css-1xdhyk6"
      href="https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
      <span
        className="css-0"
      >
        This link opens in a new window
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="link-privacy-policy-external-icon"
        role="img"
        title="External link"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </a>
  </div>
  <div
    className="chakra-stack newsletter-signup-form css-tl3ftk"
  >
    <form
      aria-label="form"
      className="css-1xdhyk6"
      id="newsletter-form"
      onSubmit={[MockFunction]}
    >
      <div
        className="css-fadkce"
        id="newsletter-form-parent"
      >
        <div
          className="css-l5w1vc"
          id="newsletter-form-child0"
        >
          <div
            className="css-1u8qly9"
            id="email-input-wrapper"
          >
            <label
              className="css-1xdhyk6"
              htmlFor="email-input"
              id="email-input-label"
            >
              Email Address
              <span>
                 (Required)
              </span>
            </label>
            <div
              className="css-79elbk"
            >
              <input
                aria-describedby="email-input-helperText"
                aria-required={true}
                autoComplete="email"
                className="chakra-input css-0"
                disabled={true}
                id="email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your email address"
                required={true}
                step={null}
                type="email"
                value=""
              />
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={false}
              id="email-input-helperText"
            >
              <div
                className="css-0"
              >
                Ex: 
                jdoe@domain.com
                <br />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-l5w1vc"
          id="newsletter-form-child1"
        >
          <button
            className="chakra-button css-1xdhyk6"
            disabled={true}
            id="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
